src/p5.dance.js

```diff
@@ -356,42 +356,49 @@ module.exports = class DanceParty {
 
 // Dance Moves
 
-  changeMoveLR(sprite, move, dir) {
-    if (!this.spriteExists_(sprite)) {
-      return;
-    }
+  getNewChangeMove(requested_change, current_move) {
     // Number of valid full length moves
     const { fullLengthMoveCount, restMoveCount } = this.world;
     const firstNonRestingMoveIndex = restMoveCount;
     // The "rest" moves are assumed to always be at the beginning
     const nonRestingFullLengthMoveCount = fullLengthMoveCount - restMoveCount;
+    if (nonRestingFullLengthMoveCount <= 1) {
+      throw "next/prev/rand requires that we have 2 or more non-resting full length moves";
+    }
+    let move = current_move;
+    if (requested_change === "next") {
+      move = current_move + 1;
+      if (move >= fullLengthMoveCount) {
+        move = firstNonRestingMoveIndex;
+      }
+    } else if (requested_change === "prev") {
+      move = current_move - 1;
+      if (move < firstNonRestingMoveIndex) {
+        move = fullLengthMoveCount - 1;
+      }
+    } else if (requested_change === "rand") {
+      // Make sure random switches to a new move
+      while (move === current_move) {
+        move = randomInt(this.world.restMoveCount, this.world.fullLengthMoveCount - 1);
+      }
```

Comment:
One way to do this and avoid a while loop is to come up with a random number between 1 and the totalLength - 1, and then increment by that number.

i.e. if you want a random number in the range (0, 10) and your current index is i, you would do
```
length = 10; // 11 - 1
iNext = (i + randomInt(1, length)) % length
```

The math to get this approach working for your scenario would be a little more difficult, since the bottom of your range is non-zero, but it could be an option if you wanted to avoid the while loop (which I always find a little scary).